From e4bd3442c2223802bf8c4a4d868e3b9443c7caf4 Mon Sep 17 00:00:00 2001
From: Jouni Malinen <jouni.malinen@oss.qualcomm.com>
Date: Mon, 18 May 2026 12:13:21 +0300
Subject: [PATCH] MLD: Verify link ID validity in MLE in reconfiguration cases

Ignore or reject invalid link ID value 15 in ML reconfiguration.

Signed-off-by: Jouni Malinen <jouni.malinen@oss.qualcomm.com>
[The src/ap/ieee802_11_eht.c part is split out into
 openwrt-patches/services/hostapd/patches/r93-0014-MLD-Verify-link-ID-validity-in-MLE-in-reconfiguratio.patch,
 as hostapd_parse_link_reconf_req_sta_profile() is only added by
 r93-0003. The wpa_supplicant/wnm_sta.c part is dropped,
 wnm_parse_neighbor_report_multi_link() does not exist in this
 snapshot.]
---
 wpa_supplicant/bss.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/wpa_supplicant/bss.c b/wpa_supplicant/bss.c
index 8727827..f5db162 100644
--- a/wpa_supplicant/bss.c
+++ b/wpa_supplicant/bss.c
@@ -2002,7 +2002,8 @@ u16 wpa_bss_parse_reconf_ml_element(struct wpa_supplicant *wpa_s,
 			u8 link_id;
 
 			link_id = control & EHT_PER_STA_RECONF_CTRL_LINK_ID_MSK;
-			removed_links |= BIT(link_id);
+			if (link_id < MAX_NUM_MLD_LINKS)
+				removed_links |= BIT(link_id);
 		}
 
 		pos += 2 + sub_elem_len;
-- 
2.34.1

